scripts/enable_fips.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+# Install the necessary software and rebuild GRUB if we're instructed to enable FIPS support.
+if [[ "$ENABLE_FIPS_MODE" == "true" ]]; then
+  # install and enable fips modules
+  sudo yum install -y dracut-fips openssl
+  sudo dracut -f
+
+  # enable fips in the boot command
+  sudo sed -i 's/^\(GRUB_CMDLINE_LINUX_DEFAULT=.*\)"$/\1 fips=1"/' /etc/default/grub
+
+  # rebuild grub
+  sudo grub2-mkconfig -o /etc/grub2.cfg
+fi
+
+sudo reboot
```

Comment:
Why are we rebooting regardless of if we make a change or not?

Reply:
Copy+paste error, this was taken from the changes originally made to the `upgrade_kernel.sh` script which reboots at the end. I think it would be best to keep this reboot, remove the one in `upgrade_kernel.sh` and move the `enable_fips.sh` script to run just after the `upgrade_kernel.sh` script.

Reply:
or maybe a separate provisioning step to perform the reboot would be cleaner?

Reply:
We definitely don't want to keep this reboot, or it might do nothing and then reboot. Given that it's a FIPs specific script, we don't want to hide anything in there like the reboot. However, I don't have an objection to creating a task that just reboots and putting all things that require reboots before it. That probably makes sense if there's multiple steps where a reboot is needed.